Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Calibration/EditorExtrinsicsCalibration.cs

```diff
@@ -0,0 +1,305 @@
+
+            networkingService = NetworkingService as INetworkingService;
+            if (networkingService != null)
+            {
+                networkingService.DataReceived += OnDataReceived;
+            }
+
+            matchMakingService = MatchMakingService as IMatchMakingService;
+            if (matchMakingService != null)
+            {
+                matchMakingService.Connect();
+            }
+
+            if (headsetCalibration != null)
+            {
+                headsetCalibration.Updated += OnHeadsetCalibrationUpdated;
+            }
+
+            for(int i = 0; i < nextArUcoImageId; i++)
+            {
+                var dslrTexture = CalibrationDataHelper.LoadDSLRArUcoImage(i);
+                var headsetData = CalibrationDataHelper.LoadHeadsetData(i);
+
+                if (dslrTexture == null ||
+                    headsetData == null ||
+                    !ProcessArUcoData(headsetData, dslrTexture))
+                {
+                    Debug.LogWarning($"Failed to process dataset: {i}");
+                }
+                else
+                {
+                    CalibrationDataHelper.SaveDSLRArUcoDetectedImage(dslrTexture, i);
+                    CreateVisual(headsetData, i);
+                }
+            }
+        }
+
+        private void OnHeadsetCalibrationUpdated(byte[] data)
+        {
+            if(HeadsetCalibrationData.TryDeserialize(data, out var headsetData))
+            {
+                dataQueue.Enqueue(headsetData);
+            }
+        }
+
+        private void Update()
+        {
+            if (feedImage != null &&
+                feedImage.texture == null)
+                feedImage.texture = CompositorWrapper.GetDSLRFeed();
+
+            if (Input.GetKeyDown(KeyCode.Space))
+            {
+                if (networkingService != null &&
+                    matchMakingService != null)
+                {
+                    var request = new HeadsetCalibrationDataRequest();
+                    request.timestamp = Time.time;
+                    var payload = request.Serialize();
+
+                    if (networkingService.SendData(payload, NetworkPriority.Critical))
+                    {
+                        Debug.Log($"Sent headset calibration data request to HoloLens at {request.timestamp}");
+                    }
+                    else
+                    {
+                        Debug.LogWarning("Failed to send headset calibration data request to HoloLens");
+                    }
+                }
+
+                if (headsetCalibration != null)
+                {
+                    Debug.Log("Requesting headset calibration data from VR Headset");
+                    headsetCalibration.UpdateHeadsetCalibrationData();
+                }
+            }
+
+            while (dataQueue.Count > 0)
+            {
+                var data = dataQueue.Dequeue();
+
+                if (dataQueue.Count == 0)
```

Comment:
If we're only ever going to use the most recently queued thing's data, then we probably shouldn't store a queue - just store the last thing.